File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.js
function solution(n) {
    let Fib = [];
    const Big = 1234567;
    
    Fib[0] = 0;
    Fib[1] = 1;
    
    for (let i = 2; i <= n; i++) {
        Fib[i] = (Fib[i - 1] + Fib[i - 2]) % Big;
    }
    
    return Fib[n];
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    fib = [0] * (n + 1)
    
    fib[0], fib[1] = 0, 1
    
    for i in range(2, n + 1):
        fib[i] = (fib[i - 2] + fib[i - 1]) % 1234567
    
    return fib[n]